### DISCUSSION TITLE
Install as a Service for MacOS via MDM

### DISCUSSION THREAD

**Discussion by ahughes-gfcu** [rustdesk/rustdesk#8206] [General]
I have been working unsuccessfully at attempting to write a bash script to install rustdesk as a service on macos that can be deployed via MDM (Intune in my example). the repo for the two scripts ive tried and a slightly modified daemon.plist are here. https://github.com/ahughes-gfcu/rustdesk_macos_serviceinstall/

The issue I'm running into is that while the com.carriez.RustDesk_service service starts, the com.carriez.RustDesk_server does not. and this seems to cause an additional problem where the Applications button on the users side will stop working and give you this error.
https://cdn.discordapp.com/attachments/804630702657110018/1242209235853639753/Screenshot_2024-05-20_at_4.14.45_PM.png?ex=665835c0&is=6656e440&hm=65221f9d00e55ff033adb98bd6002332d1e616cc873b4ac896dc024ec5f986e4&

at first I tried using the agent.plist and daemon.plist in the repo itself under src/platform/privileges_scripts. the agent works without issue but trying to get the daemon.plist to run via launchctl doesnt. I also noticed that when you install rustdesk as a service using the GUI button, the plist it installs is different than the one listed in the repo, but even changing the plist didnt seem to fix the issue. 

I would like to purchase rustdesk pro, but I need to be able to deploy it as a service without end user input.

**Comment by C-Duv** [comment]
Did you managed to remote install+start RustDesk service via MDM?

I've been trying to do the same thing and it kinda work (using Ansible, not MDM) but I fail to start/stop the service and application on demand (when I need to update the configuration).

I don't understand the goal of both daemon and agent plist files.

**Reply by ahughes-gfcu** [reply]
I was never able to get this to deploy and work as intended unfortunately.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/discussions/7762
